### PR TITLE
Increase tracing for ProfileOptimization_CheckFileExists

### PR DESCRIPTION
**Pull request by danmosemsft**
Relates to https://github.com/dotnet/corefx/issues/31792

I don't have a theory since I am not familiar with this API. 

Making sure it can write successfully to the same file used, and also seeing whether the file appears within 5 seconds if it does not exist immediately.

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@danmosemsft thank's for looking into this...weird...i'm corious to to understand what's going on :confused:.

File: src/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.netcoreapp.cs [hunk 1]
```diff
@@ -4,41 +4,51 @@
...
-                ProfileOptimization.StartProfile(Path.GetFileName(profileFilePath));
+                // Delete the file and verify the delete
+                File.Delete(_profileFile);
+                Assert.True(!File.Exists(_profileFile), $"'{_profileFile} ought to not exist now");
+
+                // Perform the test work
+                ProfileOptimization.SetProfileRoot(Path.GetDirectoryName(_profileFile));
+                ProfileOptimization.StartProfile(Path.GetFileName(_profileFile));
+
+            }, profileFile).Dispose();
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.netcoreapp.cs]: just for curiosity , does the profiling runs in the same process as remote invoke or a different one ?

Comment (danmosemsft) [reply src/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.netcoreapp.cs]: I don’t know what the API does!

Comment (MarcoRossignoli) [reply src/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.netcoreapp.cs]: FYI @Anipik https://github.com/dotnet/corefx/pull/31810#issuecomment-413980343

**Pull request merged**